game/src/game_state.rs

```diff
@@ -336,17 +346,61 @@ impl GameState {
                 do_lobby_game_init(&info).await;
             }
             forge::end_wnd_proc();
-            app_socket::send_message(&mut ws_send, "/game/start", ()).await
+            app_socket::send_message(&mut ws_send, "/game/start", ())
+                .await
                 .map_err(|_| GameInitError::Closed)?;
 
             let start_game_request = GameThreadRequestType::StartGame;
             let game_done = send_game_request(&game_request_send, start_game_request);
             game_done.await;
             let results = results.await?;
-            app_socket::send_message(&mut ws_send, "/game/end", results).await
+            app_socket::send_message(&mut ws_send, "/game/end", &results)
+                .await
                 .map_err(|_| GameInitError::Closed)?;
+
+            // Attempt to send results to the server, if this fails, we expect
+            // the app to retry in the future
+            let client = reqwest::Client::new();
```

Comment:
Oh one more thing: reqwest seems to default for no timeout, maybe we should specify something here? Though I'd expect there's still some timeout value at TCP level?

Reply:
This is a great point, went with what seems to be typical for request timeouts in browsers (30 seconds)